test/Microsoft.TestPlatform.Client.UnitTests/Discovery/DiscoveryRequestTests.cs

```diff
@@ -91,16 +91,35 @@ public void AbortIfDiscoveryRequestDisposedShouldThrowObjectDisposedException()
         Assert.ThrowsException<ObjectDisposedException>(() => _discoveryRequest.Abort());
     }
 
-    [TestMethod]
-    public void AbortIfDiscoveryIsinProgressShouldCallDiscoveryManagerAbort()
+    [DataTestMethod]
+    [DataRow(0)]
+    [DataRow(1)]
+    [DataRow(2)]
+    [DataRow(3)]
+    [DataRow(4)]
+    [DataRow(5)]
```

Comment:
Do we have a version 6? Should we add it here?

Reply:
Yes! 

There are also 2 private fields that hold the highest supported version of protocol, but I don't think they have to be separate. We should make them one constant, and add a test that makes sure we are testing up to (and including) that number.

Not in this PR though.

Reply:
V6 is the new default and is handled in the test below (named `AbortIfDiscoveryIsinProgressShouldAndSufficientProtocolVersionCallsDiscoveryManagerAbortThis`). I was thinking about having some kind of range test that goes from v6 to "current" version but I don't know how to make it with mstest. I could do a loop within the method but it's less easy to debug/understand which version is/are failing.

Reply:
We can bind data with `DynamicData` attribute?

```csharp
[DynamicData(nameof(VersionProvider), DynamicDataSourceType.Property)]
public void TestMethod1(int version, bool expected)
{
    Assert.AreEqual(expected, actual != 3);
}

public static IEnumerable<object[]> VersionProvider
{
    get
    {
        yield return new object[] { 1, true };
        yield return new object[] { 2, true };
        yield return new object[] { 3, false };
    }
}
```

Reply:
I have addressed your feedback, see https://github.com/microsoft/vstest/pull/3349#discussion_r809119036